src/migrations/20230330130554-goal-duplicate-cleanup.js

```diff
@@ -0,0 +1,2281 @@
+                
+            FROM "PreCountStatsByRegion" pre
+            JOIN "PostCountStatsByRegion" post ON pre."regionId" = post."regionId"
+        )
+        SELECT * FROM "RegionDiffs";
+        
+        WITH "CollectStats" AS (
+            SELECT 1 id, *,
+                (SELECT SUM("GoalsTotal") FROM "PreCountStatsByRegion" WHERE "regionId" = -1) AS pre_count
+            FROM "GoalStats"
+            UNION
+            SELECT 2 id, *,
+                (SELECT SUM("ActivityReportGoalsTotal") FROM "PreCountStatsByRegion" WHERE "regionId" = -1) AS pre_count
+            FROM "ActivityReportGoalStats"
+            UNION
+            SELECT 3 id, *,
+                (SELECT SUM("ActivityReportGoalResourcesTotal") FROM "PreCountStatsByRegion" WHERE "regionId" = -1) AS pre_count
+            FROM "ActivityReportGoalResourceStats"
+            UNION
+            SELECT 4 id, *,
+                (SELECT SUM("ObjectivesTotal") FROM "PreCountStatsByRegion" WHERE "regionId" = -1) AS pre_count
+            FROM "ObjectiveStats"
+            UNION
+            SELECT 5 id, *,
+                (SELECT SUM("ObjectiveFilesTotal") FROM "PreCountStatsByRegion" WHERE "regionId" = -1) AS pre_count
+            FROM "ObjectiveFileStats"
+            UNION
+            SELECT 6 id, *,
+                (SELECT SUM("ObjectiveResourcesTotal") FROM "PreCountStatsByRegion" WHERE "regionId" = -1) AS pre_count
+            FROM "ObjectiveResourceStats"
+            UNION
+            SELECT 7 id, *,
+                (SELECT SUM("ObjectiveTopicsTotal") FROM "PreCountStatsByRegion" WHERE "regionId" = -1) AS pre_count
+            FROM "ObjectiveTopicStats"
+            UNION
+            SELECT 8 id, *,
+                (SELECT SUM("ActivityReportObjectivesTotal") FROM "PreCountStatsByRegion" WHERE "regionId" = -1) AS pre_count
+            FROM "ActivityReportObjectiveStats"
+            UNION
+            SELECT 9 id, *,
+                (SELECT SUM("ActivityReportObjectiveFilesTotal") FROM "PreCountStatsByRegion" WHERE "regionId" = -1) AS pre_count
+            FROM "ActivityReportObjectiveFileStats"
+            UNION
+            SELECT 10 id, *,
+                (SELECT SUM("ActivityReportObjectiveResourcesTotal") FROM "PreCountStatsByRegion" WHERE "regionId" = -1) AS pre_count
+            FROM "ActivityReportObjectiveResourceStats"
+            UNION
+            SELECT 11 id, *,
+                (SELECT SUM("ActivityReportObjectiveTopicsTotal") FROM "PreCountStatsByRegion" WHERE "regionId" = -1) AS pre_count
+            FROM "ActivityReportObjectiveTopicStats"
+        )
+        SELECT *, 
+            pre_count - post_count AS diff,
+            post_count - (pre_count - "Deletes" + "Inserts") AS adjusted_diff
+
+        FROM "CollectStats"
+        ORDER BY id;
+        DROP TABLE "PreCountStatsByRegion" ;
+          `, { transaction });
+      } catch (err) {
```

Comment:
We need to drop all the temp tables in the script either right before they are created or at the end. for some reason temp tables stick around during back to back migration reseeds and will cause issues.

Reply:
Are you thinking that RegionDiffs need also have a temp table create/delete step?
 ```sql
line 24: 
DROP TABLE IF EXISTS "PreCountStatsByRegion";
CREATE TEMP TABLE "PreCountStatsByRegion" AS (
...

line 2155
DROP TABLE IF EXISTS "PostCountStatsByRegion";
CREATE TEMP TABLE "PostCountStatsByRegion" AS (

extra drop:
line 2271  
DROP TABLE "PreCountStatsByRegion" ;
```

Reply:
One of the other PR's inflight will allow reseeding the DB. the result of this is that any migration file using temp tables needs to guard for the temp table already existing. So please add drop if exists before each creation of temp tables.